### PR TITLE
Move CI setup to GitHub Actions

### PR DESCRIPTION
**Pull request by foolip**
Travis is quite slow to start builds.

The Travis setup also ran on pull requests, although it doesn't show up
in the .travis.yml file.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    branches:
+      - master
+  pull_request: {}
+name: test
+jobs:
+  test:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-node@v1
```
Comment (KyleJu) [inline .github/workflows/test.yml]: why is it setup-node here? Maybe actions/setup-go?

Comment (foolip) [reply .github/workflows/test.yml]: Oh, big oops :)

**Comment by KyleJu** [comment]
Currently experiencing issues related to https://github.com/actions/setup-go/issues/12

**Comment by KyleJu** [comment]
@foolip Added a temporary fix suggested in the thread above. Seems to be working!

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      with:
+        go-version: 1.11.x
+    - name: Set GOPATH
+      # temporary fix
+      # see https://github.com/actions/setup-go/issues/12
+      # see https://github.com/actions/setup-go/issues/14
+      run: |
+        echo "##[set-env name=GOPATH;]$(dirname $GITHUB_WORKSPACE)"
+        echo "##[add-path]$(dirname $GITHUB_WORKSPACE)/bin"
+      shell: bash
```
Comment (foolip) [inline .github/workflows/test.yml]: Bash is the default on Linux and macOS, so you can omit this.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (foolip) [inline .github/workflows/test.yml]:
```suggestion
            curl -s https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```

For less noisy output.

**Comment by foolip** [comment]
@KyleJu your changes look good, thanks for making this work! I left a few comments, feel free to resolve and merge this when you're confident it's in good shape.

**Pull request merged**